oak-lucene/src/main/java/org/apache/jackrabbit/oak/plugins/index/lucene/spi/IndexFieldProvider.java

```diff
@@ -25,6 +25,9 @@
 /**
  * Implementations of this interface would get callbacks while indexing documents. It's the responsibility
  * of the implementation to exit as early as possible if it doesn't care about the document being indexed.
+ *
+ * @deprecated This interface exposes Lucene API directly - it will be removed soon.
+ * Use the "dynamicBoost" feature instead.
  */
 public interface IndexFieldProvider {
```

Comment:
We should add @Deprecated annotation to the interface as well along with the one in comment
@Deprecated
public interface FulltextQueryTermsProvider {